Backend/router/v1/report.ctrl.js

```diff
@@ -1,26 +1,50 @@
 const models = require('../../models');
 
 const createReport = (req, res) => {
-  if (req.isAuthenticated()) {
+  if (!req.isAuthenticated()) {
+    res.status(403);
+    res.json({
+      msg : "you are not logined"
+    });
+    res.end();
+    return ;
+  }
+
+  models.perpetrator.findOrCreate({
+    where: {
+      facebook_url: req.body.data[0].facebook_url
+    }
+  }).then((result) => {
     models.report.create({
       what: req.body.data[0].what,
       location: req.body.data[0].location,
       time: req.body.data[0].time,
       who: req.body.data[0].who,
       details: req.body.data[0].details,
+      perpetratorID: result[0].id,
       userID: req.user[0].dataValues.id
     }).then((result) => {
-      res.status(201);
-      res.json({
-        msg: "Report is made successfully"
-      });
-    });
-  } else {
-    res.status(403);
-    res.json({
-      msg : "You are not logged in"
+      models.report.count({
+        group: ['userID', 'perpetratorID'],
+        attributes: ['userID', 'perpetratorID'],
+        where: {
+          perpetratorID: result.perpetratorID
+        }
+      }).then((count) => {
+          models.perpetrator.update({
+            count: count.length
```

Comment:
don't need +1, because this code count the certain perpetrator's reported num. And delete same user's report num. Maybe what I should  is put "-1" because we have to return except the user's own.

Reply:
Okay, looks like we need to add comments here. It's not clear just by looking at the code.

Reply:
Shouldn't you be updating the count for delete/update report use cases?

Reply:
oh I will write the issue for eunyoung!

Reply:
Please add a TODO